crates/swc_ecma_transforms_optimization/src/simplify/branch/mod.rs

```diff
@@ -108,7 +108,10 @@ impl VisitMut for Remover {
                 if s.exprs.is_empty() {
                     *e = Expr::dummy();
                 } else if s.exprs.len() == 1 {
-                    *e = *s.exprs.pop().unwrap();
+                    // TODO: preserve PURE comment attached to original
+                    // sequence expression.
+
+                    // *e = *s.exprs.pop().unwrap();
```

Comment:
You can

1. assign the span of the node with `PURE` comment to the emitted node

2. Accept `C: Comments` or `&dyn Comments` and call `move_leading`